test/definitions/scenarios/capsule_upgrade_test.rb

```diff
@@ -0,0 +1,168 @@
+require 'test_helper'
+
+describe "capsule upgrade scenarios" do
+  include DefinitionsTestHelper
+
+  before(:each) do
+    assume_feature_present(:capsule)
+    ForemanMaintain.config.stubs(:manage_crond).returns(true)
+  end
+
+  describe Scenarios::Satellite::PreUpgradeCheck do
+    let(:scenario) do
+      Scenarios::Satellite::PreUpgradeCheck.new
+    end
+
+    it 'composes all steps for Satellite on EL8' do
+      Scenarios::Satellite::PreUpgradeCheck.any_instance.stubs(:el_major_version).returns(8)
+
+      assert_scenario_has_steps(
+        scenario,
+        Checks::ServerPing,
+        Checks::ServicesUp,
+        Checks::SystemRegistration,
+        Checks::CheckHotfixInstalled,
+        Checks::CheckTmout,
+        Checks::CheckUpstreamRepository,
+        Checks::Disk::AvailableSpace,
+        Checks::NonRhPackages,
+        Checks::PackageManager::Dnf::ValidateDnfConfig,
+        Checks::Repositories::CheckNonRhRepository,
+        Checks::CheckIpv6Disable,
+        Checks::Disk::AvailableSpacePostgresql13,
+        Checks::CheckOrganizationContentAccessMode,
+        Checks::Repositories::Validate,
+      )
+    end
+
+    it 'composes all steps for Satellite on EL9' do
+      Scenarios::Satellite::PreUpgradeCheck.any_instance.stubs(:el_major_version).returns(9)
+
+      assert_scenario_has_steps(
+        scenario,
+        Checks::ServerPing,
+        Checks::ServicesUp,
+        Checks::SystemRegistration,
+        Checks::CheckHotfixInstalled,
+        Checks::CheckTmout,
+        Checks::CheckUpstreamRepository,
+        Checks::Disk::AvailableSpace,
+        Checks::NonRhPackages,
+        Checks::PackageManager::Dnf::ValidateDnfConfig,
+        Checks::Repositories::CheckNonRhRepository,
+        Checks::CheckIpv6Disable,
+        Checks::Disk::AvailableSpacePostgresql13,
+        Checks::CheckOrganizationContentAccessMode,
+        Checks::Repositories::Validate,
+      )
+    end
+  end
+
+  describe Scenarios::Satellite::PreMigrations do
+    let(:scenario) do
+      Scenarios::Satellite::PreMigrations.new
+    end
+
+    it 'composes all steps for Satellite on EL8' do
+      Scenarios::Satellite::PreMigrations.any_instance.stubs(:el_major_version).returns(8)
+
+      assert_scenario_has_steps(
+        scenario,
+        Procedures::MaintenanceMode::EnableMaintenanceMode,
+        Procedures::Crond::Stop,
+      )
+    end
+
+    it 'composes all steps for Satellite on EL9' do
+      Scenarios::Satellite::PreMigrations.any_instance.stubs(:el_major_version).returns(9)
+
+      assert_scenario_has_steps(
+        scenario,
+        Procedures::MaintenanceMode::EnableMaintenanceMode,
+        Procedures::Crond::Stop,
+      )
+    end
+  end
+
+  describe Scenarios::Satellite::Migrations do
+    let(:scenario) do
+      Scenarios::Satellite::Migrations.new
+    end
+
+    it 'composes all steps for Capsule on EL8' do
```

Comment:
When I wrote this, I only looked at "enable modules", but in reality there are more steps:
```
        add_step(Procedures::Repositories::Setup.new(:version => target_version))
        if el8?
          modules_to_switch = ['postgresql:13']
          add_step(Procedures::Packages::SwitchModules.new(:module_names => modules_to_switch))
          modules_to_enable = ["#{feature(:instance).downstream.module_name}:#{el_short_name}"]
          add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable))
        end
        add_step(Procedures::Packages::Update.new(
          :assumeyes => true,
          :download_only => true
        ))
        add_step(Procedures::Service::Stop.new)
        add_step(Procedures::Packages::Update.new(:assumeyes => true, :clean_cache => false))
        add_step_with_context(Procedures::Installer::Run)
        add_step(Procedures::Installer::UpgradeRakeTask)
```

mind adding these?

Reply:
Doh - I added these in the Satellite tests but forgot in the Capsule one. Will do.